src/serialization/UntypedNode.cs

```diff
@@ -0,0 +1,35 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Kiota.Abstractions.Serialization
+{
+    /// <summary>
+    /// Base class for untyped node.
+    /// </summary>
+    public class UntypedNode : IParsable
+    {
+        /// <summary>
+        /// The value assigned to untyped node.
+        /// </summary>
+        public object? Value => null;
+        /// <summary>
+        /// The deserialization information for the current model
+        /// </summary>
+        public virtual IDictionary<string, Action<IParseNode>> GetFieldDeserializers() => new Dictionary<string, Action<IParseNode>>();
```

Comment:
this will return a new dictionary every time it's called.
Could we assign a ReadOnly Dictionary to a constant once, and return that constant instead to avoid unecessary allocs?

Reply:
Good point. Fixed.